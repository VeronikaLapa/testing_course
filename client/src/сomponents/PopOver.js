import React, {useEffect} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import fetchUser from "../store/actions/fetchUser";
import getAuth from "../store/actions/getAuth";

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        flexGrow: 1,
    },
}));

function PopOver({auth, name, getAuthUserAction}) {
    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState(null);
    const [open, setOpen] = React.useState(false);

    useEffect(() => {
        if (anchorEl && auth) {
            setOpen(true);
        } else {
            setOpen(false);
        }
        if (!auth) {
            setAnchorEl(null);
        }
    });

    const handleMenu = (event) => {
        console.log('handle menu');
        setAnchorEl(event.currentTarget);
        if (name == null) {
            //getAuthUserAction();
        }
        setOpen(true);

    };

    const handleClose = () => {
        setOpen(false);
        setAnchorEl(null);
    };


    return (
        <div>
            {auth && (
                <div>
                    <IconButton
                        aria-label="account of current user"
                        aria-controls="menu-appbar"
                        aria-haspopup="true"
                        onClick={handleMenu}
                        color="inherit"
                        id='avatar'
                    >
                        <AccountCircle/>
                    </IconButton>
                    <Menu
                        id="menu-appbar"
                        anchorEl={anchorEl}
                        anchorOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        keepMounted
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        open={open}
                        onClose={handleClose}
                    >
                        <MenuItem onClick={handleClose}>{name}</MenuItem>
                        <MenuItem><Link to="/logout"><Typography color="error">Exit</Typography></Link></MenuItem>
                    </Menu>
                </div>
            )}
        </div>
    );
}

const mapStateToProps = store => {
    console.log(store); // посмотрим, что же у нас в store?
    return {
        auth: store.user.auth,
        name: store.user.user.name
    }
};
const mapDispatchToProps = dispatch => {
    return {
        getAuthUserAction: () => dispatch(getAuth()), // [1]
    }
};


export default connect(mapStateToProps, mapDispatchToProps)(PopOver);
