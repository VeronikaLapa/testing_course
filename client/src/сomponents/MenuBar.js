import React, {useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import PopOver from "./PopOver";

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        flexGrow: 1,
    },
}));

function MenuBar({auth}) {

    const classes = useStyles();
    return (
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" className={classes.title}>
                        MY SNet
                    </Typography>
                    <PopOver />
                </Toolbar>
            </AppBar>
        </div>
    );
}


export default MenuBar;
