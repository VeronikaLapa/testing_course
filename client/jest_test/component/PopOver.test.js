import configureStore from 'redux-mock-store'
import MenuBar from "../../src/сomponents/MenuBar";
import React from "react";
import renderer from 'react-test-renderer'
import {Provider} from "react-redux";
import {MemoryRouter, Router} from "react-router-dom";
import {cleanup} from "@testing-library/react";
import PopOver from "../../src/сomponents/PopOver";

describe('PopOver render without auth',()=> {
    const initialState = {user:{auth: false, user: {name: ''}}};
    const mockStore = configureStore();
    let store, container;

    afterEach(cleanup);

    beforeEach(() => {
        store = mockStore(initialState);
    });

    it('Render popover without auth', () => {
        const renderedValue = renderer.create(<MemoryRouter><Provider store={store}><PopOver /></Provider></MemoryRouter>).toJSON();
        expect(renderedValue).toMatchSnapshot();
        expect(renderedValue.childList).toBeEmpty;
    });
});
