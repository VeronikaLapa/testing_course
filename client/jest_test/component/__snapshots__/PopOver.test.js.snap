// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopOver render without auth Render popover 1`] = `<div />`;

exports[`PopOver render without auth Render popover without auth 1`] = `<div />`;
